scrapy/pipelines/media.py

```diff
@@ -206,7 +206,8 @@ def _cache_result_and_execute_waiters(
             # minimize cached information for failure
             result.cleanFailure()
             result.frames = []
-            result.stack = []
+            if twisted.version.major < 24 or (twisted.version.major == 24 and twisted.version.minor < 10):
```

Comment:
You shoul simplify the comparison using `Version`: https://github.com/scrapy/scrapy/blob/master/scrapy/mail.py#L223

Reply:
##Changed it to:
   ```python
            if twisted_version <= Version("twisted", 24, 10, 0):
                result.stack = []
```